testing/trino-tests/src/test/java/io/trino/execution/TestTableRedirection.java

```diff
@@ -0,0 +1,349 @@
+
+        assertQuery(TEST_SESSION,
+                format("SELECT table_schema, table_name, column_name FROM information_schema.columns WHERE table_schema = '%s' AND table_name='%s'", SCHEMA_ONE,
+                        REDIRECTION_LOOP_PING),
+                "SELECT '', '', '' WHERE 1 = 0");
+    }
+
+    @Test
+    public void testShowCreate()
+    {
+        String showCreateValidSource = getSingleValueString(format("SHOW CREATE TABLE %s.%s", SCHEMA_ONE, VALID_REDIRECTION_SRC));
+        String showCreateValidTarget = getSingleValueString(format("SHOW CREATE TABLE %s.%s", SCHEMA_TWO, VALID_REDIRECTION_TARGET));
+        assertEquals(
+                showCreateValidTarget,
+                showCreateValidSource.replace(SCHEMA_ONE + "." + VALID_REDIRECTION_SRC, SCHEMA_TWO + "." + VALID_REDIRECTION_TARGET));
+
+        assertThatThrownBy(() -> query((format("SHOW CREATE TABLE %s.%s", SCHEMA_ONE, BAD_REDIRECTION_SRC))))
+                .hasMessageContaining(format(
+                        "Table '%s' has been redirected to '%s'. Table '%s' does not exist.",
+                        new CatalogSchemaTableName(CATALOG_NAME, SCHEMA_ONE, BAD_REDIRECTION_SRC),
+                        new CatalogSchemaTableName(CATALOG_NAME, SCHEMA_TWO, NON_EXISTENT_TABLE),
+                        new CatalogSchemaTableName(CATALOG_NAME, SCHEMA_TWO, NON_EXISTENT_TABLE)));
+
+        assertThatThrownBy(() -> query((format("SHOW CREATE TABLE %s.%s", SCHEMA_ONE, REDIRECTION_LOOP_PING))))
+                .hasMessageContaining("Table redirections form a loop");
+    }
+
+    private String getSingleValueString(String query)
+    {
+        return (String) getOnlyElement(computeActual(query).getMaterializedRows()).getField(0);
+    }
+
+    @Test
+    public void testShowColumns()
+    {
+        assertQuery(TEST_SESSION,
+                format("SHOW COLUMNS FROM %s.%s", SCHEMA_ONE, VALID_REDIRECTION_SRC),
+                "VALUES "
+                        + row(C2, BIGINT.getDisplayName(), "", "") + ","
+                        + row(C3, BIGINT.getDisplayName(), "", ""));
+
+        assertThatThrownBy(() -> query((format("SHOW COLUMNS FROM %s.%s", SCHEMA_ONE, BAD_REDIRECTION_SRC))))
+                .hasMessageContaining(format(
+                        "Table '%s' has been redirected to '%s'. Table '%s' does not exist.",
+                        new CatalogSchemaTableName(CATALOG_NAME, SCHEMA_ONE, BAD_REDIRECTION_SRC),
+                        new CatalogSchemaTableName(CATALOG_NAME, SCHEMA_TWO, NON_EXISTENT_TABLE),
+                        new CatalogSchemaTableName(CATALOG_NAME, SCHEMA_TWO, NON_EXISTENT_TABLE)));
+
+        assertThatThrownBy(() -> query((format("SHOW COLUMNS FROM %s.%s", SCHEMA_ONE, REDIRECTION_LOOP_PING))))
+                .hasMessageContaining("Table redirections form a loop");
+    }
+
+    // TODO: Add tests for redirection in CommentsSystemTable and CREATE TABLE LIKE
+
+    private static String row(String... values)
```

Comment:
maybe move this to AbstractTestQueryFramework

Reply:
have kept this in the test class for now as a quick utility, as ATQF has more generic methods around query assertions. I can generalize this method and do a followup if you feel strongly.